lib/Dialect/LLHD/Transforms/ProcessLoweringPass.cpp

```diff
@@ -60,10 +71,14 @@ void ProcessLoweringPass::runOnOperation() {
         // Every probed signal has to occur in the observed signals list in
         // the wait instruction
         WalkResult result = op.walk([&wait](llhd::PrbOp prbOp) -> WalkResult {
-          if (!llvm::is_contained(wait.obs(), prbOp.signal())) {
-            return wait.emitOpError(
-                "Process-lowering: The wait terminator is required to have "
-                "all probed signals as arguments!");
+          SmallVector<Value, 8> signals;
+          getAllMuxedSignals(prbOp.signal(), signals);
+          for (Value sig : signals) {
+            if (!llvm::is_contained(wait.obs(), sig)) {
+              return wait.emitOpError(
+                  "Process-lowering: The wait terminator is required to have "
+                  "all probed signals as arguments!");
```

Comment:
Just nitpicking: I think the primary error messages are supposed to start with a lowercase latter, something like:
```suggestion
                  "process-lowering: the wait terminator is required to have "
                  "all probed signals as arguments");
```

Reply:
I will address this in a follow-up commit to keep this functional addition separat from cleanup.